x/gamm/types/route_test.go

```diff
@@ -0,0 +1,26 @@
+package types_test
+
+import (
+	"testing"
+
+	proto "github.com/gogo/protobuf/proto"
+	"github.com/stretchr/testify/require"
+
+	gammtypes "github.com/osmosis-labs/osmosis/v13/x/gamm/types"
+	swaproutertypes "github.com/osmosis-labs/osmosis/v13/x/swaprouter/types"
+)
+
+// TestSwapRoutesSerialization tests that while swap routes
+// are proto-generated from different modules, they are identical in terms
+// of serialization and deserialization.
+func TestSwapRoutes_MarshalUnmarshal(t *testing.T) {
+	swapRouterExactAmountInRoute := swaproutertypes.SwapAmountInRoute{}
+	gammExactAmountInRoute := gammtypes.SwapAmountInRoute{}
+
+	swapRouterBz, err := proto.Marshal(&swapRouterExactAmountInRoute)
+	require.NoError(t, err)
+	gammBz, err := proto.Marshal(&gammExactAmountInRoute)
+	require.NoError(t, err)
+
+	require.Equal(t, swapRouterBz, gammBz)
+}
```

Comment:
What's the reason we do not delete `SwapAmountInRoute` from gamm types completely? Seems that `SwapAmountInRoute` in gamm types have no actual usage and its not getting called anywhere

Reply:
We can delete that in a subsequent PR. It is currently being used as proof that there should be no breakage

Reply:
This doesn't guarantee things, we need non-empty entries for every field for this to actually be checked.

Reply:
Missed adding values. Added now

Reply:
https://github.com/osmosis-labs/osmosis/issues/3921